physionet-django/physionet/test_gcs.py

```diff
@@ -0,0 +1,184 @@
+        gcs_object.client.create_bucket('test')
+        gcs_object.mkdir()
+
+        # WHEN + THEN
+        self.assertRaises(GCSObjectException, gcs_object.mkdir)
+
+    def test_size_when_object_is_file(self):
+        # GIVEN
+        gcs_object = self._monkeypatch_gcsobject(GCSObject('test/dir1/notes.txt'))
+        gcs_object.client.create_bucket('test')
+        gcs_object.upload_from_string('content')
+
+        # WHEN + THEN
+        self.assertEqual(gcs_object.size(), len('content'))
+
+    def test_size_when_object_is_directory(self):
+        # GIVEN
+        gcs_object = self._monkeypatch_gcsobject(GCSObject('test/dir1/'))
+        gcs_object_1 = self._monkeypatch_gcsobject(GCSObject('test/dir1/notes1.txt'))
+        gcs_object_2 = self._monkeypatch_gcsobject(GCSObject('test/dir1/notes2.txt'))
+
+        # create a bucket
+        gcs_object.client.create_bucket('test')
+
+        # put files into a bucket
+        gcs_object_1.upload_from_string('content')
+        gcs_object_2.upload_from_string('content')
+
+        # WHEN + THEN
+        self.assertEqual(gcs_object.size(), len('content') * 2)
+
+    def test_rm_deletes_all_files_in_directory_when_object_is_directory(self):
+        # GIVEN
+        gcs_object = self._monkeypatch_gcsobject(GCSObject('test/dir1/'))
+        gcs_object_1 = self._monkeypatch_gcsobject(GCSObject('test/dir1/notes1.txt'))
+        gcs_object_2 = self._monkeypatch_gcsobject(GCSObject('test/dir1/notes2.txt'))
+
+        # create a bucket
+        gcs_object.client.create_bucket('test')
+
+        # put files into a bucket
+        gcs_object_1.upload_from_string('content')
+        gcs_object_2.upload_from_string('content')
+
+        # WHEN
+        gcs_object.rm()
+
+        # THEN
+        self.assertEqual(gcs_object.size(), 0)
+
+    def test_rm_removes_file_when_object_is_file(self):
+        # GIVEN
+        gcs_object = self._monkeypatch_gcsobject(GCSObject('test/dir/file.jpg'))
+        gcs_object.client.create_bucket('test')
+        gcs_object.upload_from_string('content')
+
+        # WHEN
+        gcs_object.rm()
+
+        # THEN
+        dir_ = self._monkeypatch_gcsobject(self._monkeypatch_gcsobject(GCSObject('test/dir/')))
+        self.assertEqual(dir_.size(), 0)
+
+    def test_cp_copies_file_to_directory(self):
+        # GIVEN
+        gcs_object = self._monkeypatch_gcsobject(GCSObject('test/dir/file.jpg'))
+        gcs_object_1 = self._monkeypatch_gcsobject(GCSObject('test/dir/'))
+
+        # create a bucket
+        gcs_object.client.create_bucket('test')
+
+        # put a file into a bucket
+        gcs_object.upload_from_string('content')
+
+        # WHEN
+        gcs_object_1.cp(self._monkeypatch_gcsobject(GCSObject('test/dir_copied/')))
+
+        # THEN
+        self.assertEqual(gcs_object_1.size(), len('content'))
+        self.assertEqual(gcs_object.size(), len('content'))
+
+    def test_mv_moves_file_when_object_is_file(self):
```

Comment:
I think it's a good idea to add more tests for moving files (when source an target is a file, when target is equal to `..` etc.)